src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/MouseInputActionObserver.cs

```diff
@@ -0,0 +1,183 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Newtonsoft.Json;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace StateRecorder
+{
+    [Serializable]
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    public class MouseInputActionData : InputActionData
+    {
+        // non-fractional pixel accuracy
+        public int[] position;
+
+        public bool[] leftMiddleRightForwardBackButton;
```

Comment:
When I was looking at this data in the Python notebook, I found it a little strange that these were packed in to an array. Is there a particular reason for that? It seems like it would be a little more ergonomic to just have separate left/middle/right/forward/back button booleans, and I don't believe there's any reason it would be less performant.

(Non-blocking; this should work fine, mostly just curious as to the choice)

Reply:
I originally had it that way in the json.

I moved to this not as a performance optimization but more for my sanity when reading the data.  When click dragging the mouse you can get up to 30 mouse events per tick, which led to pages and pages of scrolling trying to read them all in VS code :/ 

IOW... there is no real performance or technical reason for it to be one way vs the other (maybe the zipped json is ever so slightly large with the explicit keys, but not enough to write home about).  It is this way solely so that I could read multiple mouse events on the screen at the same time when reviewing the data.

Reply:
I guess my concern would be that it's less clear to program against. Intuitively I was expecting something more like `inputs.mouse[0].leftButtonDown`.

Still, it's fine either way, as long as it's clear.